pinecone/models.go

```diff
@@ -81,35 +97,57 @@ type PodSpec struct {
 	MetadataConfig   *PodSpecMetadataConfig `json:"metadata_config,omitempty"`
 }
 
+// ServerlessSpec is the infrastructure specification of a serverless Pinecone Index. Only available for serverless Indexes.
 type ServerlessSpec struct {
 	Cloud  Cloud  `json:"cloud"`
 	Region string `json:"region"`
 }
 
+// Vector is a [dense or sparse vector object] with optional metadata.
+//
+// [dense or sparse vector object]: https://docs.pinecone.io/guides/get-started/key-concepts#dense-vector
 type Vector struct {
 	Id           string        `json:"id"`
 	Values       []float32     `json:"values,omitempty"`
 	SparseValues *SparseValues `json:"sparse_values,omitempty"`
 	Metadata     *Metadata     `json:"metadata,omitempty"`
 }
 
+// ScoredVector is a vector with an associated similarity score calculated according to the distance metric of the
+// Index.
 type ScoredVector struct {
 	Vector *Vector `json:"vector,omitempty"`
 	Score  float32 `json:"score"`
 }
 
+// SparseValues is a sparse vector objects, most commonly used for [hybrid search].
+//
+// [hybrid search]: https://docs.pinecone.io/guides/data/understanding-hybrid-search#hybrid-search-in-pinecone
 type SparseValues struct {
 	Indices []uint32  `json:"indices,omitempty"`
 	Values  []float32 `json:"values,omitempty"`
 }
 
+// NamespaceSummary is a summary of stats for a Pinecone [namespace].
+//
+// [namespace]: https://docs.pinecone.io/guides/indexes/use-namespaces
 type NamespaceSummary struct {
 	VectorCount uint32 `json:"vector_count"`
 }
 
+// Usage is the usage stats ([Read Units]) for a Pinecone Index.
+//
+// [Read Units]: https://docs.pinecone.io/guides/organizations/manage-cost/understanding-cost#serverless-indexes
 type Usage struct {
 	ReadUnits uint32 `json:"read_units"`
 }
 
+// Filter is the structure that holds the conditions for a query sent to a Pinecone Index,
+// e.g. 'give me all vectors where "author" is "John"'.
 type Filter = structpb.Struct
```

Comment:
I think this type and the `Metadata` type may need clarification.

`Filter` is used primarily in Querying, Deleting, and Describing Index Stats:
- https://github.com/pinecone-io/go-pinecone/blob/cef7a46b73f9e6ce62f4c3eaf7ebb75be548a7c6/pinecone/index_connection.go#L397
- https://github.com/pinecone-io/go-pinecone/blob/cef7a46b73f9e6ce62f4c3eaf7ebb75be548a7c6/pinecone/index_connection.go#L514
- https://github.com/pinecone-io/go-pinecone/blob/cef7a46b73f9e6ce62f4c3eaf7ebb75be548a7c6/pinecone/index_connection.go#L703
- https://github.com/pinecone-io/go-pinecone/blob/cef7a46b73f9e6ce62f4c3eaf7ebb75be548a7c6/pinecone/index_connection.go#L956

`Metadata` is what's used in `Vector`, so upsert and other data plane operations. Now I'm sort of curious why there's two types representing the same thing, maybe we should consolidate? We should do that in a follow-up PR, I don't think you necessarily need to make that change here.

Reply:
I think they're different, actually.... 

- In `index_connection`, `Metadata` is the metadata you want to attach to a vector, e.g. in [UpdateVectorRequest](https://github.com/pinecone-io/go-pinecone/blob/main/pinecone/index_connection.go#L776)
- While `Filter` is the metadata filter you want to apply to a request, e.g. in [QueryByVectorValuesRequest](https://github.com/pinecone-io/go-pinecone/blob/main/pinecone/index_connection.go#L397). 

Maybe we're saying the same thing?

Reply:
[We connected offline; we are going to rename `Filter` to `MetadataFilter` in a follow-up PR.]